### ISSUE TITLE
Rename test packages

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#965]
Testing package names are not consistent:

- [x] `sdk/metric/processor/test`
- [x] `sdk/metric/aggregator/test`
- [x] `sdk/metric/controller/test`
- [x] `exporters/metric/test`
- [ ] `internal/testing`
- [x] `api/testharness`
- [x] `api/trace/testtrace`

Unify these package names to follow the standard library naming convention of `{package name}test`. E.g.

- `io/iotest`
- `http/httptest`

So the above listed would become:

- `sdk/metric/processor/processortest`
- `sdk/metric/aggregator/aggregatortest`
- `sdk/metric/controller/controllertest`
- ~`exporters/metric/metrictest`~
- `internal/internaltest`\*
- `api/apitest`
- `api/trace/tracetest`

[\*] The `internal/testing` package should likely be investigated to see if it can be renamed or merged into something more reflective of the purpose it serves. The `internal/metric` package also includes testing mocks that could be related. Otherwise, renaming to `internal/internaltest` would be preferred if it cannot be restructured.

**Comment by MrAlias** [comment]
The `exporters/metric/test` package is handled here: https://github.com/open-telemetry/opentelemetry-go/pull/962

**Comment by eundoosong** [comment]
Hello, I'd love to take this as first contribution.
Can I modify all? It looks there are many places to be modified.

**Comment by MrAlias** [comment]
@eundoosong yeah, you'll need to update all the test packages to complete this issue. How you want to structure that in PRs and commits is up to you. Just ask that it is presented for review in a somewhat clear and digestable form.

Also, you'll want to wait on https://github.com/open-telemetry/opentelemetry-go/pull/977 to land as it cleans up some of the use of the `sdk/metric/processor/test`.

**Comment by MrAlias** [comment]
@eundoosong it might also be a good approach to look into what needs to be done with `internal/testing` and post here if you have any questions prior to making the change.

**Comment by eundoosong** [comment]
@MrAlias 
Thanks, 
I'll group testing package changes into each PR so that the review can be in digestable form.
For `internal/testing`, I'll look into while I'm working on the other.

**Comment by MrAlias** [comment]
@eundoosong are you still able to work on the reset of the renamings?

**Comment by eundoosong** [comment]
> @eundoosong are you still able to work on the reset of the renamings?

Sorry, I have been busy for another task at work.
Yes, I have a plan to work on the rest till this week.  :)

**Comment by eundoosong** [comment]
@MrAlias 
Hello, I'd like to finish off this issue and I have questions.

> The internal/testing package should likely be investigated to see if it can be renamed or merged into something more reflective of the purpose it serves. The internal/metric package also includes testing mocks that could be related. Otherwise, renaming to internal/internaltest would be preferred if it cannot be restructured.

After some investigation in `internal/testing` code, I'm starting to wonder what the point is why this package needs to be(or can be..) restructured.
This package includes `byte alignment check`, `env storage for test`, `error for test` that are widely used for opentelemetry-go's testing, do you think those can be moved to somewhere? or should be separate into its own package?

**Comment by eundoosong** [comment]
@MrAlias  Ping

**Comment by MrAlias** [comment]
@eundoosong sorry about the delay. Sounds like just renaming `internal/testing` to `internal/internaltest` would be ideal if it cannot be refactored.
